Source/WebCore/crypto/mac/CryptoKeyECMac.cpp

```diff
@@ -90,48 +96,90 @@ static constexpr bool doesFieldElementMatchNamedCurve(CryptoKeyEC::NamedCurve cu
 
 size_t CryptoKeyEC::keySizeInBits() const
 {
-    int result = CCECGetKeySize(m_platformKey.get());
-    return result ? result : 0;
+
+    return WTF::switchOn( m_platformKey, [&](const std::unique_ptr<typename std::remove_pointer<CCECCryptorRef>::type, WebCore::CCECCryptorRefDeleter>& options) ->int {
+        return CCECGetKeySize(options.get());
+        },
+        [](const Vector<uint8_t>& options) -> int {
+            return options.size()*8;
+    });
+
+
+}
+PlatformECKey CryptoKeyEC::platformKey() const
+{
+    return WTF::switchOn( m_platformKey, [&](const std::unique_ptr<typename std::remove_pointer<CCECCryptorRef>::type, WebCore::CCECCryptorRefDeleter>& options) -> PlatformECKey {
```

Comment:
This creates a lot of switchOn.
Should we not use a separate class for E25519 keys?

Reply:
We use variants if we can share some code.
Can you tell which code is actually shared?
If we are not sharing any code between the two, it might make the code more complex to understand without real benefits.

Reply:
Extra space after `WTF::switchOn(`